pumi/pumi_ghost.cc

```diff
@@ -0,0 +1,612 @@
+        adj_ent.clear();
+        pumi_ment_get2ndAdj (ghost_ent, brg_dim, ghost_dim, adj_ent);
+        //std::cout<<"("<<pumi_rank()<<") "<<__func__<<": pumi_ment_get2ndAdj.size()="<<adj_ent.size()<<"\n";
+
+        for (std::vector<pMeshEnt>::iterator git=adj_ent.begin(); git!=adj_ent.end(); ++git)
+        {
+          if (m->isGhost(*git) || m->hasTag(*git,tag))
+            continue; // skip ghost copy or already-processed copy
+      
+          //copy_RC_to_BP(brg_ent, ghost_ent);
+          APF_ITERATE(apf::Copies,remotes,rit)
+            plan->send(*git, rit->first);
+
+          m->setIntTag(*git,tag,&dummy);
+          processed_ent.push_back(*git);
+          ++num_prev_layer;
+        } // for (std::vector<pMeshEnt>::iterator git=adj_ent.begin()
+      } // for int i=start_prev_layer
+      start_prev_layer+=size_prev_layer;
+      size_prev_layer+=num_prev_layer;
+    } // for layer
+    for (std::vector<pMeshEnt>::iterator git=processed_ent.begin(); git!=processed_ent.end(); ++git)
+      m->removeTag(*git,tag);
+//    std::cout<<"("<<pumi_rank()<<") "<<__func__<<": "<<__LINE__<<" brg_ent "<<pumi_ment_getGlobalID(brg_ent)<<"\n";
+  } // while brg_ent
+  m->end(it);
+  m->destroyTag(tag);
+
+// ********************************************
+// STEP 2: perform ghosting
+// ********************************************
+//  std::cout<<"("<<pumi_rank()<<") "<<__func__<<": plan->count()="<<plan->count()<<"\n";
+  if (!PCU_Comm_Self())
+    printf("ghosting plan computed in %f seconds\n", PCU_Time()-t0);
+
+  pumi_ghost_create(m, plan);
+
+
+}
+
+// *********************************************************
+void pumi_ghost_delete (pMesh m)
+// *********************************************************
+{
+  pMeshTag tag = pumi::instance()->ghosted_tag;
+  if (!tag) return;
+
+  for (int d=3; d>=0; --d)
+  {
+    for (std::vector<pMeshEnt>::iterator it=pumi::instance()->ghost_vec[d].begin();
+         it!=pumi::instance()->ghost_vec[d].end(); ++it)
+      m->destroy(*it);
+    for (std::vector<pMeshEnt>::iterator it=pumi::instance()->ghosted_vec[d].begin();
+         it!=pumi::instance()->ghosted_vec[d].end(); ++it)
+    {
+      m->removeTag(*it,tag);
+      m->deleteGhost(*it);
+    }
+  }
+
+  // delete tag
+  m->destroyTag(pumi::instance()->ghost_tag);
+  pumi::instance()->ghost_tag = NULL;
+  m->destroyTag(pumi::instance()->ghosted_tag);
+  pumi::instance()->ghosted_tag = NULL;
+
+  for (int d=3; d>=0; --d)
+  {
+    pumi::instance()->ghost_vec[d].clear();
+    pumi::instance()->ghosted_vec[d].clear();
+  }
+}
+
+// *********************************************************
+void pumi_ghost_getInfo (pMesh, std::vector<int>&)
+// *********************************************************
+{
+  if (!pumi_rank()) 
+    std::cout<<"[PUMI ERROR] "<<__func__<<" failed: not supported\n";
```

Comment:
Should this (and other unsupported functions) be removed until they are available?